.github/workflows/deploy.yml

```diff
@@ -0,0 +1,55 @@
+name: Deploy
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+jobs:
+  deploy-staging:
+    name: Deploy staging.wpt.fyi
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_IMAGE: webplatformtests/wpt.fyi:latest
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+
+      - name: secrets
+        run: echo "$GCLOUD_KEY_FILE_JSON" > client-secret.json
+        env:
+         GCLOUD_KEY_FILE_JSON: ${{ secrets.GCLOUD_KEY_FILE_JSON }}
+
+      - name: Set DOCKER_INSTANCE
+        run: echo "DOCKER_INSTANCE=wptd-dev-$(echo $RANDOM)" >> $GITHUB_ENV
+
+      - name: pre-installation
+        run: |
+          docker pull "${DOCKER_IMAGE}"
+          bash ./util/docker-dev/run.sh -d -q
+
+      - name: installation
+        run: docker exec -t "${DOCKER_INSTANCE}" make go_build;
+
+      # Set -f for main branch.
+      - name: set deployment flag
+        if: ${{ github.ref == 'refs/heads/main' }}
+        run: echo "FORCE_DEPLOYMENT=-f" >> $GITHUB_ENV
+
+      - name: build webapp
+        run: ./util/deploy-staging.sh "${FORCE_DEPLOYMENT}" webapp/web/app.staging.yaml
+
+      - name: Run go_large_test in main
```

Comment:
Should these tests really be run as part of the deploy job, such that if they fail the webapp deployment (above) happens but the rest (below) does not?

Perhaps this can be split into its own workflow and only run on PRs? Alternatively, if passing certain tests is important before deploying, it would be good to make that more explicit by running the tests first.

Reply:
`go_large_test` appears to be the webdriver integration tests that need staging deployment. In the [origin travis setup](https://github.com/web-platform-tests/wpt.fyi/blob/207813b3ed18bae81068934caa478daffd782d36/.travis.yml#L61), they only run in the deployment of main branch. Maybe it is too costly to run in PRs?